### DISCUSSION TITLE
🚀 gym-ignition and ScenarIO v1.0.0 Released

### DISCUSSION THREAD

**Discussion by diegoferigo** [robotology/community#445] [Releases]
Hi @robotology/everyone,

Today the first stable version of **gym-ignition** was released and is now available at https://github.com/robotology/gym-ignition/releases/tag/v1.0.0.

**gym-ignition** is a project that targets both control and robot learning research domains:

- Researchers in robotics and control can simulate their robots with familiar tools like Gazebo and URDF, without the need to rely on any middleware.
- Researchers in robot learning can quickly develop new robotic environments that can scale to hundreds of parallel instances.

For the first target, we developed a new abstraction layer called **SCENe InterfAces for Robot Input/Output** . This first major version freezes its APIs and provides a full implementation for the new [Ignition Gazebo](https://ignitionrobotics.org/) simulator. ScenarIO is developed in C++ and we officially provide a `scenario` Python package containing the SWIG bindings. If you are familiar with similar tools, the Gazebo ScenarIO implentation can be compared to other popular alternatives as [pybullet](https://github.com/bulletphysics/bullet3) and [mujoco-py](https://github.com/openai/mujoco-py). You can find more details in the [Why ScenarIO](https://robotology.github.io/gym-ignition/master/motivations/why_gym_ignition.html#) section of the website.

For the second target, we developed **`gym_ignition`**, a pure Python package that, by calling `scenario`, streamlines the development of OpenAI Gym compatible robotic environments for reinforcement learning research. It centralizes all the boilerplate code that otherwise would be copied and pasted in each environment, and provides a set of classes and helpers based on [iDynTree](https://github.com/robotology/idyntree/) with useful robotic algorithms. You can find more details in the [Why gym-ignition](https://robotology.github.io/gym-ignition/master/motivations/why_gym_ignition.html#why-gym-ignition) section of the website.

Being the very first stable release of the project, there might still be bugs and problems in the configuration of your system. Please notify us by [opening an issue](https://github.com/robotology/gym-ignition/issues), we'll try to address it as soon as possible.

Where to start now? [Install the project](https://robotology.github.io/gym-ignition/master/installation/support_policy.html) and then head off to the getting started section:

- [Simulation of a simple pendulum](https://robotology.github.io/gym-ignition/master/getting_started/scenario.html).
- [Pick and place with a Panda manipulator](https://robotology.github.io/gym-ignition/master/getting_started/manipulation.html).

||||
|:---:|:---:|:---:|
| ![](https://user-images.githubusercontent.com/469199/99262383-321fb200-281e-11eb-89cc-cc31f590daa3.png) | ![](https://user-images.githubusercontent.com/469199/99263111-0cdf7380-281f-11eb-9cfe-338b2aae0503.png) | ![](https://user-images.githubusercontent.com/469199/99262746-9e021a80-281e-11eb-9df1-d70134b0801a.png) |

**Comment by HorvathDawson** [comment]
Hey @diegoferigo I found this project while trying to find a good option for my undergraduate capstone aiming to do RL sim2real on a planarized monopod. I spent the last few days trying to work out how to get it running and I keep getting issues when the worlds physics engine gets set. 

Using the same code as the simple pendulum from the getting started section on the website I get an error finding the dart physics engine.

`[Err] [Physics.cpp:542] Failed to find plugin [ignition-physics-dartsim-plugin]` 

I looked where the error came from and it was from an empty path from finding the shared library. I set the property the same as in the example, 

` world.set_physics_engine(scenario_gazebo.PhysicsEngine_dart)`

When the GUI loads the pendulum is in the world but it never ends up swinging down. I get the same error trying the manipulator example.

I tried looking through your documentation but didn't see anything about missing any environment variables etc. I would love to get help setting this up! Thank you for everything you have done with this!

I am using Ignition-dome installed with binaries and the stable install of gym-ignition

**Reply by diegoferigo** [reply]
Hi @HorvathDawson, thanks for reporting. Can you please open an issue in the [gym-ignition](https://github.com/robotology/gym-ignition) repository? Copying and pasting what you wrote here would suffice. I'll try to have a look over the weekend.
